app/core/views.py

```diff
@@ -0,0 +1,174 @@
+
+class AnswerViewSet(viewsets.ModelViewSet):
+    queryset = AnswerModel.objects.all()
+    serializer_class = AnswerSerializer
+    permission_classes = [IsAdminUser]
+
+class CategoryQuestionViewSet(viewsets.ModelViewSet):
+    queryset = CategoryQuestionModel.objects.all()
+    serializer_class = CategoryQuestionSerializer
+    permission_classes = [IsAdminUser]
+
+
+class RankingViewSet(
+    mixins.ListModelMixin, mixins.RetrieveModelMixin, viewsets.GenericViewSet
+):
+    queryset = RankingModel.objects.all().order_by('-value')
+    serializer_class = RankingSerializer
+    permission_classes = [PlayerPermission]
+    
+    # Neste caso, pk eh o id da category.
+    def retrieve(self, request, pk):
+        ranking = (
+            RankingModel.objects.filter(category_id__in=[pk]).order_by('-value')
+        )
+        serializer = RankingSerializer(ranking, many=True)
+        return Response(serializer.data)
+        
+
+class QuizAPIView(APIView):
+    permission_classes = [PlayerPermission]
+    def get(self, requests, category_id):
+        cursor = connection.cursor()
+        query = f'SELECT B.id, B.question FROM  core_categoryquestionmodel as A INNER JOIN core_questionmodel as B ON B.id = A.question_id WHERE A.category_id="{category_id.replace("-","")}";'
+        cursor.execute(query)
+        questions_db = cursor.fetchall()
+        questions = [
+            {'id': UUID(x), 'question': y} for x, y in questions_db
+        ]
+        random.shuffle(questions)
+        answers_db = AnswerModel.objects.all()
+        answers = list(answers_db.values())
+        list_quiz = []
+        for question in questions:
+            random.shuffle(answers)
+            question_answers_true = list(
+                filter(
+                    lambda x: (
+                        x['question_id'] == question['id']
+                        and x['correct_answer'] is True
+                    ),
+                    answers
+                )
+            )[0:1]
+            question_answers_false = list(
+                filter(
+                    lambda x: (
+                        x['question_id'] == question['id']
+                        and x['correct_answer'] is False
+                    ),
+                    answers
+                )
+            )[0:2]
+            list_questions = question_answers_true + question_answers_false
+            random.shuffle(list_questions)
+            data = {
+                'question': {
+                    'id':question['id'],
+                    'question': question['question']
+                },
+                'answers': [
+                    {'id': x['id'], 'answer': x['answer']}
+                    for x in list_questions
+                ]
+            }
+            list_quiz.append(data)
+        # Verifica se o quiz possui >= 10 perguntas.
+        if len(list_quiz) >= 10:
+            return Response(list_quiz[0:10], status.HTTP_200_OK)
```

Comment:
Fazendo essa consulta logo após a consulta você evitaria o processamento desnecessário do código acima. Opte sempre por efetuar as validações no início do código.

Reply:
Perfeito, também concordo com você. Isso evitaria este processamento. Se tivemos executando em lambda da AWS onde o tempo é crucial, perderíamos um tempo desnecessário fazendo a validação perto do retorno. Mas, acho que isso foi uma implementação de melhoria, por isso ficou mau colocada.